# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and compare with the data printed above.


2. How many observations (rows) are in total?


3. How many variables (columns) are in total? What do they represent?


4. How many observatoins are for each type of flower? 


5. What is the type of data for each variable?


6. What are the units of each variable?

In [32]:
# 1
import pandas as pd

ruta = "datasets/iris/iris.csv"
dataset = pd.read_csv(ruta)

In [33]:
dataset

setalL  setalW  petalL  petalW           Class
0       5.1     3.5     1.4     0.2     Iris-setosa
1       4.9     3.0     1.4     0.2     Iris-setosa
2       4.7     3.2     1.3     0.2     Iris-setosa
3       4.6     3.1     1.5     0.2     Iris-setosa
4       5.0     3.6     1.4     0.2     Iris-setosa
..      ...     ...     ...     ...             ...
145     6.7     3.0     5.2     2.3  Iris-virginica
146     6.3     2.5     5.0     1.9  Iris-virginica
147     6.5     3.0     5.2     2.0  Iris-virginica
148     6.2     3.4     5.4     2.3  Iris-virginica
149     5.9     3.0     5.1     1.8  Iris-virginica

[150 rows x 5 columns]

# 2
Existen 149 lineas de observacion

# 3
Son 5 columnas, representadas respectivamente por longitud de seta, ancho de seta, longitud de petalo, ancho de petalo, todo en cm, y clase

In [34]:
dataset.Class.value_counts()

Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: Class, dtype: int64

# 4
Existen 50 observaciones de cada tipo de flor

In [35]:
dataset.columns

Index(['setalL', 'setalW', 'petalL', 'petalW', 'Class'], dtype='object')

In [36]:
dataset.dtypes

setalL    float64
setalW    float64
petalL    float64
petalW    float64
Class      object
dtype: object

# 5
El tipo para los datos en cm son tipo float, mientras la clase es tipo object

# 6
La unidad de las variables son centimetros